x-pack/plugins/lens/public/indexpattern_datasource/dimension_panel/dimension_editor.scss

```diff
@@ -15,6 +15,10 @@
   column-gap: $euiSizeXL;
 }
 
+.lnsIndexPatternDimensionEditor__operation .euiListGroupItem__label {
```

Comment:
It kind of looks like this shipped has sailed already in `dimension_editor.tsx` but we're not supposed to be overriding `.eui*` class names and should instead be supplying our own custom ones (which EUI components generally accept). 

Any chance for doing the same here?

Reply:
Yeah, that's right. We should be passing a class name to the list group items.

Reply:
Whenever I can, I do pass the class :) I am unfortunately not able to achieve what I want with adding classes because element that I have to style cannot accept extra class (`euiListGroup__label`). If someone has better idea, let me know 🙏 
This is what I am trying to achieve:
![image](https://user-images.githubusercontent.com/4283304/99559235-d8a5b780-29c4-11eb-804f-377df722a4ed.png)

To test it, modify the line in the file: https://github.com/elastic/kibana/pull/83437/files#diff-e350db2049bff2db77fe5191ee0af1f6bc078d50d87795948db206c0a179b80dR161

 `if (disabledStatus) {` to `if(true){`

Reply:
After poking around at this a bit, I better understand the situation - this is a nested/generated element that cannot be directly targeted with a classname. Aside from that issue, the structure here is not semantic HTML as there are divs inside of a button element.

To address both of these issue, it seems best to pull the icon/tooltip out of the label and use the `extraAction` prop. However, that actually adds a separate button which is not exactly what we want here either.

Long story short, I will create an EUI issue requesting that the `extraAction` prop be more flexible. In this case, allowing the use of an `EuiIconTip`. I'll report back with a link if you would kindly add it as a comment in your code for later follow up. Thanks!

Reply:
Here is the aforementioned issue: https://github.com/elastic/eui/issues/4285

In the meantime, I don't know if it's possible to go with the label only and remove all the flex item elements, but that would avoid us producing invalid markup.

Reply:
Thank you @ryankeairns, this is very helpful! 🙇‍♀️

Reply:
_My 2 cents, and then I'll jump back out._ 

The tooltip and info icon here are indications for why the current item is disabled. Adding the info icon actually pulls more attention to the things you **cannot** do versus the things you can. It also restricts getting this info to a 24x24px area instead of the whole disabled item. Instead, I'd suggest removing the icon and placing the tooltip around the entire list item. So only when users think "Huh, why is this grayed out" do they interact to find out the reason, but the list doesn't get cluttered with info icons that pull attention to those disabled items.

Info icons are much better to indicate actually providing more info like explaining what the functions are an not for indicating/explaining why it's in a certain state.

A caveat to that would be if the item has an **error**. For instance, if the **currently selected** function doesn't work with the field, then you'd really want to call attention to it with the `alert` icon and a tooltip like so:

<img width="347" alt="Screen Shot 2020-11-19 at 15 18 35 PM" src="https://user-images.githubusercontent.com/549577/99719652-98087580-2a7a-11eb-982f-9540f4c0733e.png">

Which is possible with `iconType`. But what's not available is changing the tooltip content. We might be able to add that in EUI.

Reply:
> The tooltip and info icon here are indications for why the current item is disabled. Adding the info icon actually pulls more attention to the things you **cannot** do versus the things you can. It also restricts getting this info to a 24x24px area instead of the whole disabled item. Instead, I'd suggest removing the icon and placing the tooltip around the entire list item. So only when users think "Huh, why is this grayed out" do they interact to find out the reason, but the list doesn't get cluttered with info icons that pull attention to those disabled items.

This is great feedback, @cchaos. Thanks for pass it along. Looking at it again from your perspective, I agree with your recommendation. @mbondyra, would it be possible to ignore the icon I recommended in my design and just have the tooltip wrap the entire list item as Caroline has suggested? Thanks!